File: backend/internal/models/task.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Task struct {
	ID          string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	UserID      string    `gorm:"not null;type:varchar(36);index" json:"user_id"`
	Title       string    `gorm:"not null;type:varchar(255)" json:"title"`
	Description string    `gorm:"type:text" json:"description"`
	Status      string    `gorm:"type:varchar(20);default:'pending'" json:"status"` // pending, in_progress, completed
	DueDate     *time.Time `gorm:"type:timestamp" json:"due_date"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

File: lib/domain/entities/task.dart
class Task {
  final String id;
  final String userId;
  final String title;
  final String? description;
  final String status; // pending, in_progress, completed
  final DateTime? dueDate;
  final DateTime createdAt;
  final DateTime updatedAt;

  Task({
    required this.id,
    required this.userId,
    required this.title,
    this.description,
    required this.status,
    this.dueDate,
    required this.createdAt,
    required this.updatedAt,
  });
}






